File: app/src/main/java/com/yuriysurzhikov/autobroker/AutoBrokerApp.java
package com.yuriysurzhikov.autobroker;

import android.app.Application;

import com.yuriysurzhikov.autobroker.repository.core.ISynchronizer;

import javax.inject.Inject;

import dagger.hilt.android.HiltAndroidApp;

@HiltAndroidApp
public class AutoBrokerApp extends Application {

    @Inject
    public ISynchronizer sSynchronizer;

    @Override
    public void onCreate() {
        super.onCreate();
        sync(sSynchronizer);
    }

    public static void sync(ISynchronizer sSynchronizer) {
        sSynchronizer.performSync();
    }
}

File: app/src/main/java/com/yuriysurzhikov/autobroker/AutoBrokerApp.kt
package com.yuriysurzhikov.autobroker

import android.app.Application
import android.content.Context
import android.os.Bundle
import com.yuriysurzhikov.autobroker.repository.sync.SyncAdapter
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class AutoBrokerApp : Application() {

    override fun onCreate() {
        super.onCreate()
        sync(this, null)
    }

    companion object {

        @JvmStatic
        fun sync(context: Context, bundle: Bundle?) {
            SyncAdapter(context).performSync(bundle)
        }
    }
}